### ISSUE TITLE
[dev] Different behavior of "Accept-Encoding" header

### ISSUE THREAD

**Issue by autarc** [cloudflare/wrangler#1425]
Original [mentioned here as feedback](https://github.com/cloudflare/wrangler/issues/1047#issuecomment-651872430) for the `dev` mode.
___

```js
request.headers.get('Accept-Encoding');
```

Currently the `wrangler dev` runtime has different behavior for the `Accept-Encoding` header than a cloud deployed worker. Accessing it returns the actual values of a client (e.g. `gzip, deflate, br`) which can be misleading as later on [those won't be forwarded but always replaced with `gzip`](https://support.cloudflare.com/hc/en-us/articles/200168086-Does-Cloudflare-compress-resources-). Having them aligned helps to be consistent so developers can be aware of that circumstance before running the worker in production.

**Comment by EverlastingBugstopper** [comment]
ok - from my investigation when sending `Accept-Encoding: gzip, deflate, br` as a request header

```js
addEventListener('fetch', event => {
  event.respondWith(handleRequest(event.request))
})

async function handleRequest(request) { 
  let accept_encoding = request.headers.get("Accept-Encoding")
  console.log(accept_encoding)
  let response = await fetch("https://example.com")
  let content_encoding = response.headers.get("Content-Encoding")
  console.log(content_encoding)
  return response
}

```

|command|`console.log(accept_encoding)`|`console.log(content_encoding)`|observed `Content-Encoding` response header|
|---|---|---|---|
|`wrangler dev`|`gzip, deflate, br`|`null`|Header does not exist|
|`wrangler preview`|`gzip, deflate, br`|`null`|Header does not exist|
|`wrangler publish`|`gzip`|`null`|`br`|

**Comment by EverlastingBugstopper** [comment]
This will be fixed when we release the stuff in #1085 - i just tested `wrangler dev` with source built from branch `avery/edge-dev-server` and i get the following:

|command|`console.log(accept_encoding)`|`console.log(content_encoding)`|observed `Content-Encoding` response header|
|---|---|---|---|
|`wrangler dev`|`gzip`|`null`|br|

this is the same you see when you publish a worker 😄

**Comment by autarc** [comment]
The issue is resolved with the recent change of using the latest edge workers.
